### PR TITLE
use BC_ macros instead of string comparison

### PR DESCRIPTION
**Pull request by kroening**
This is less prone to error.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 3672f3038902e3c16f26a02af628951ba26b7bbf).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/115688152

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: eb61f2c342e5d166fcbd951784c101e300c55cb8).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/115688569

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 1]
```diff
@@ -1274,32 +1275,30 @@ code_blockt java_bytecode_convert_methodt::convert_instructions(
...
-    else if(statement == patternt("?load") || statement == patternt("?load_?"))
+    else if(bytecode == patternt("?load") || bytecode == patternt("?load_?"))
     {
       // load a value from a local variable
       results[0]=
         variable(arg0, statement[0], i_it->address, CAST_AS_NEEDED);
     }
-    else if(statement=="ldc" || statement=="ldc_w" ||
-            statement=="ldc2" || statement=="ldc2_w")
+    else if(bytecode == BC_ldc || bytecode == BC_ldc_w || bytecode == BC_ldc2_w)
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:1301]: ldc2?

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 2]
```diff
@@ -1342,158 +1341,160 @@ code_blockt java_bytecode_convert_methodt::convert_instructions(
...
     {
+      // clang-format off
       const irep_idt id=
-        statement=="ifeq"?ID_equal:
-        statement=="ifne"?ID_notequal:
-        statement=="iflt"?ID_lt:
-        statement=="ifge"?ID_ge:
-        statement=="ifgt"?ID_gt:
-        statement=="ifle"?ID_le:
+        bytecode==BC_ifeq ? ID_equal:
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: style

**Review by smowton — APPROVED** [review]
One possible omission, otherwise lgtm

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp (hunk 1 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:1301]: ldc2 doesn't exist, even though that spoils the symmetry.

Comment (smowton) [reply jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:1301]: Aha, so it doesn't

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 7389a8dd1f395369a3aef0d4a63f803fc39b4a90).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/115723483

**Pull request merged**